### PR TITLE
pr-pull: don't try to fetch artifacts for a pull request when the changed formulae don't use bottles

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

This should prevent errors like:

```console
% brew pr-pull 52786        
==> Fetching homebrew/core pull request #52786
Error: No artifact with the name `bottles` was found!
  https://github.com/Homebrew/homebrew-core/actions/runs/74548552
```

With this pull request, this now gives output like:

```console
% brew pr-pull --no-upload 52786
==> Fetching homebrew/core pull request #52786
==> Cherry-picking 1 commit from #52786
==> Skipping artifacts for #52786 as the formulae don't need bottles
```

### CODE REVIEW

File: Library/Homebrew/dev-cmd/pr-pull.rb [hunk 1]
```diff
@@ -114,6 +114,32 @@ def check_branch(path, ref)
...
+    Utils.popen_read("git", "-C", tap.path, "diff-tree",
+                     "-r", "--name-only", "--diff-filter=AM",
+                     original_commit, "HEAD", "--", tap.formula_dir)
+         .lines.each do |line|
+      next unless line.end_with? ".rb\n"
+
+      name = "#{tap.name}/#{File.basename(line.chomp, ".rb")}"
+      begin
+        f = Formula[name]
+      rescue Exception # rubocop:disable Lint/RescueException
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/pr-pull.rb:134]: Any chance of making this a bit more specific?

Comment (jonchang) [reply Library/Homebrew/dev-cmd/pr-pull.rb:134]: This is copied from `brew pull` and I don't really know why it's there. I'll check `git blame` to see if I can find a reason.

Comment (MikeMcQuaid) [reply Library/Homebrew/dev-cmd/pr-pull.rb:134]: Oh, no worries, just leave it in that case.

**Comment by jonchang** [comment]
Merging, thanks for the :eyes:!

**Pull request merged**